kernel-modules/probe/collector_probe.c

```diff
@@ -198,6 +218,127 @@ PROBE_SIGNATURE("sched/", sched_process_exit, sched_process_exit_args) {
   return 0;
 }
 
+////////////////////////////////////////////////////////////////////////////////////
+struct sys_enter_accept4_args {
+  __u64 pad;
+  int __syscall_nr;
+  long fd;
+  struct sockaddr* upeer_sockaddr;
+  int* upeer_len;
+  long flags;
+};
+
+struct sys_enter_chdir_args {
+  __u64 pad;
+  int __syscall_nr;
+  const char* filename;
+};
+
+struct sys_enter_fchdir_args {
+  __u64 pad;
+  int __syscall_nr;
+  long fd;
+};
+
+struct sys_enter_close_args {
+  __u64 pad;
+  int __syscall_nr;
+  long fd;
+};
+
+struct sys_enter_connect_args {
+  __u64 pad;
+  int __syscall_nr;
+  long fd;
+  struct sockaddr* uservaddr;
+  long addrlen;
+};
+
+struct sys_enter_clone_args {
+  __u64 pad;
+  int __syscall_nr;
+#ifdef __s390x__
```

Comment:
How do we keep track of these argument orderings?

Reply:
> Regarding struct packing, did we consider using a compiler directive instead of switching from `uid_t` to `unsigned int` ?

I'm aware of a compiler directive to always pack (regardless of type alignment).  But not sure how that would help in using uid_t as it would guarantee that it would pack.  Is there another directive you have in mind?

Originally, I put padding in to explicitly "unpack" the structure (and keep the uid_t and gid_t types).  But found that we relied on implicit packing elsewhere in code and with endian differences between platforms the code looked a little verbose.  I put a comment at start of structure to remind people that implicit padding is assumed.   Also, in the syscall format, the gid and uid fields actually have a size of 8 bytes whereas uid_t and gid_t are 4 byte types.  Even though we won't have an 8 byte gid, the unsigned long actually makes sense here.

Reply:
Would the use of something like `__attribute__((aligned(8))` to force alignment/padding solve this problem?

Reply:
I think it would solve half the problem.  The alignment would be good, but since syscall format is 8 bytes and our type is 4 bytes, we would need to wrap everything in a \#ifdef w/ padding to handle big/little endian differences.